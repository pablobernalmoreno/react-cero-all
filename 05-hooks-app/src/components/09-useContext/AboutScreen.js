import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const AboutScreen = () => {
  const { setUser } = useContext(UserContext);
  return (
    <div>
      <h1>AboutScreen</h1>
      <hr />
      <button className="btn btn-primary" onClick={() => setUser({})}>
        Clear
      </button>
    </div>
  );
};
