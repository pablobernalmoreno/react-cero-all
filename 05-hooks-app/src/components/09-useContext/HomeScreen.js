import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const HomeScreen = () => {
  const { user } = useContext(UserContext);
  return (
    <div>
      <h1>HomeScreen </h1>
      <h2>{JSON.stringify(user)}</h2>
      <hr />
    </div>
  );
};
