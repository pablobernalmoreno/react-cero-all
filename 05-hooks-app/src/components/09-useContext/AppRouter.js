import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { NavBar } from "./NavBar";

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <Switch>
          <Route path="/about" component={AboutScreen} />
          <Route path="/login" component={LoginScreen} />
          <Route path="/" component={HomeScreen} />
          <Route component={HomeScreen} />
        </Switch>
      </div>
    </Router>
  );
};
