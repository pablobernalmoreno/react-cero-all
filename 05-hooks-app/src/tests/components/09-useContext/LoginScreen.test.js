import { mount, shallow } from "enzyme";
import { LoginScreen } from "../../../components/09-useContext/LoginScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("Pruebas en HomeScreen", () => {
  const setUser = jest.fn();

  const wrapper = mount(
    <UserContext.Provider value={{ setUser }}>
      <LoginScreen />
    </UserContext.Provider>
  );

  test("debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de ejecutar el setUser con el argumento esperado", () => {
    const button = wrapper.find("button");
    button.simulate("click");

    expect(setUser).toHaveBeenCalledWith({
      id: 1234,
      name: "pablo",
      username: "pablito",
    });
  });
});
