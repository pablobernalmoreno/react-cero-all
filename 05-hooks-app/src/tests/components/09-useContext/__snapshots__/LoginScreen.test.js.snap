// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en HomeScreen debe de mostrarse correctamente 1`] = `
<LoginScreen>
  <div>
    <h1>
      LoginScreen
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</LoginScreen>
`;
